src/OpenSolid/Interval.elm

```diff
@@ -285,3 +324,129 @@ isContainedIn : Interval -> Interval -> Bool
+shiftBy delta (Internal.Interval { minValue, maxValue }) =
+    Internal.Interval
+        { minValue = minValue + delta
+        , maxValue = maxValue + delta
+        }
+
+
+{-| Get the image of sin(x) applied on the interval.
+
+    Interval.with { minValue = -pi / 2, maxValue = pi / 2}
+        |> Interval.sin
+    --> Interval.with { minValue = -1, maxValue = 1 }
+
+-}
+sin : Interval -> Interval
+sin ((Internal.Interval { minValue, maxValue }) as interval) =
+    if isSingleton interval then
+        singleton (Basics.sin minValue)
+    else
+        let
+            ( includesMin, includesMax ) =
+                sinIncludesMinMax interval
+
+            newMin =
+                if includesMin then
+                    -1
+                else
+                    min (Basics.sin minValue) (Basics.sin maxValue)
+
+            newMax =
+                if includesMax then
+                    1
+                else
+                    max (Basics.sin minValue) (Basics.sin maxValue)
+        in
+        Internal.Interval
+            { minValue = newMin
+            , maxValue = newMax
+            }
+
+
+{-| Get the image of cos(x) applied on the interval.
+
+    Interval.with { minValue = -pi / 2, maxValue = pi / 2}
+        |> Interval.cos
+    --> Interval.with { minValue = 0, maxValue = 1 }
+
+-}
+cos : Interval -> Interval
+cos ((Internal.Interval { minValue, maxValue }) as interval) =
+    if isSingleton interval then
+        singleton (Basics.cos minValue)
+    else
+        let
+            ( includesMin, includesMax ) =
+                cosIncludesMinMax interval
+
+            newMin =
+                if includesMin then
+                    -1
+                else
+                    min (Basics.cos minValue) (Basics.cos maxValue)
+
+            newMax =
+                if includesMax then
+                    1
+                else
+                    max (Basics.cos minValue) (Basics.cos maxValue)
+        in
+        Internal.Interval
+            { minValue = newMin
+            , maxValue = newMax
+            }
+
+
+{-| cos(x - pi/2) = sin(x), therefore if cos(interval - pi/2) includes
+the maximum/minimum, that means sin(interval) includes the maximum/minimum
+accordingly.
+-}
+sinIncludesMinMax : Interval -> ( Bool, Bool )
+sinIncludesMinMax interval =
+    interval |> shiftBy (-pi / 2) |> cosIncludesMinMax
+
+
+{-| cos(x + pi) = -cos(x), therefore if cos(interval + pi) includes the maximum,
+that means cos(interval) includes the minimum.
+-}
+cosIncludesMinMax : Interval -> ( Bool, Bool )
+cosIncludesMinMax interval =
+    ( interval |> shiftBy pi |> cosIncludesMax
+    , interval |> cosIncludesMax
+    )
+
+
+{-| The maximum of cos(x) is x = 2 pi * k for every integer k.
+If `minValue` and `maxValue` are in different branches
+(meaning diffrent values of k), then the interval must pass through
+2 pi * k, which means the interval must include the maximum value.
+-}
+cosIncludesMax : Interval -> Bool
```

Comment:
There is also the edge case when `min = 2 * k * pi` and `max < 2 * (k+1) * pi`. Both return the same `floor`. That probably could be checked by `floor ( minValue / (2*pi )) == ceiling ( minValue / (2*pi) )` ? Not sure if this is important or if it can happen.

Reply:
That's a good point, but I think in this case it's fine - if `cosIncludesMax` or one of its relatives returns `False`, then the function is evaluated at both endpoints anyways. So it should only really be important that `cosIncludesMax` catches cases that are strictly *inside* the given interval.